apis/mattermost/v1beta1/mattermost_types.go

```diff
@@ -65,23 +66,31 @@ type MattermostSpec struct {
 	// Mattermost application pods.
 	// +optional
 	Volumes []v1.Volume `json:"volumes,omitempty"`
-	// Defines additional volumeMounts to add to mattermost application pods.
+	// Defines additional volumeMounts to add to Mattermost application pods.
 	// +optional
 	VolumeMounts []v1.VolumeMount `json:"volumeMounts,omitempty"`
 
-	// Advanced contains additional advanced settings for Mattermost. These
-	// settings generally don't need to be changed.
-	Advanced Advanced `json:"advanced,omitempty"`
-
 	// External Services
 	Database      Database      `json:"database,omitempty"`
-	Filestore     Filestore     `json:"filestore,omitempty"`
+	FileStore     FileStore     `json:"fileStore,omitempty"`
 	ElasticSearch ElasticSearch `json:"elasticSearch,omitempty"`
+
+	// Advanced settings - it is recommended to leave the default configuration
+	// for below settings, unless a very specific use case arises.
+
+	// Scheduling defines the configuration related to scheduling of the Mattermost pods
+	// as well as resource constraints. These settings generally don't need to be changed.
+	// +optional
+	Scheduling Scheduling `json:"scheduling,omitempty"`
+	// Probes defines configuration of liveness and readiness probe for Mattermost pods.
+	// These settings generally don't need to be changed.
+	// +optional
+	Probes Probes `json:"probes,omitempty"`
```

Comment:
Hmm, I guess it's fine to split these out and use the comment to define them as "advanced" and likely to change, but the original design did put them in their own struct to make this a bit clearer. I am good with this, but we may want to revisit this again in the future before the beta spec is production-ready.

Reply:
I am 1/5 on that. I just did not like that the struct was called `Advanced` as it seems very vague but I couldn't find a well-fitting name that would work for all fields inside, therefore I decided to split it.
I am definitely open to discussing it further.